### ISSUE TITLE
SpinWheelSlot layout error on reopening file

### ISSUE THREAD

**Issue by billreed63** [maqetta/maqetta#3715]
With the following index.html

```
<!DOCTYPE html>
<html>
<head>
<meta charset="utf-8"/>
<title>Untitled</title>
<script type="text/javascript" src="lib/dojo/dojo/dojo.js" data-dojo-config="'parseOnLoad':true,'async':true,'packages':[{'name':'gridx','location':'../gridx'},{'name':'clipart','location':'../../clipart'},{'name':'maqettaSamples','location':'../../../samples'},{'name':'maqetta','location':'../../maqetta'},{'name':'shapes','location':'../../shapes'},{'name':'zazl','location':'../../zazl'},{'name':'widgets','location':'../../custom'}]"></script>
<script type="text/javascript">
require([
  "dijit/dijit",
  "dojo/parser",
  "maqetta/space",
  "maqetta/AppStates",
  "dojox/mobile",
  "dojox/mobile/deviceTheme",
  "dojox/mobile/compat",
  "dojo/hash",
  "dojox/mobile/ScrollableView",
  "dojox/io/xhrScriptPlugin",
  "dojo/data/ItemFileReadStore",
  "dojox/mobile/EdgeToEdgeDataList",
  "dojox/mobile/SpinWheel",
  "dojox/mobile/SpinWheelDatePicker",
  "dojox/mobile/TextArea"
]);
</script>
<style>@import "themes/claro/claro.css";@import "app.css";
</style>
<script type="text/javascript" src="app.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1.0, user-scalable=no"></meta>
</head>
<body class="claro" data-maq-device="iphone" data-maq-flow-layout="true" data-maq-ws="collapse" data-maq-comptype="mobile" id="myapp" data-maq-appstates="{}">
 <div data-dojo-type="dojox.mobile.ScrollableView" id="mainView" keepScrollPos="false" scrollBar="true">
<h1 label="Weight Tracker" data-dojo-type="dojox.mobile.Heading" fixed="top">
 <div label="+" data-dojo-type="dojox.mobile.ToolBarButton" id="addWeightButton" moveTo="detailsView" style="float: right;"></div>
</h1>
<span data-dojo-type="dojo.data.ItemFileReadStore" id="ItemFileReadStore_1" jsId="ItemFileReadStore_1" url="weights.json"></span>
<ul data-dojo-type="dojox.mobile.EdgeToEdgeDataList" id="weightList" store="ItemFileReadStore_1" query="{&quot;label&quot;:&quot;*&quot;}"></ul>
</div>
 <div data-dojo-type="dojox.mobile.ScrollableView" id="detailsView" keepScrollPos="false" scrollBar="true" selected="true">
<h1 label="Details" data-dojo-type="dojox.mobile.Heading" back="Home" moveTo="mainView" fixed="top"></h1>
  <div label="Weights (lbs)" data-dojo-type="dojox.mobile.RoundRectCategory"></div>
  <div data-dojo-type="dojox.mobile.RoundRect">
   <div data-dojo-type="dojox.mobile.SpinWheel" id="weightSpinWheel" style="width: 222px; margin-right: auto; margin-left: auto;">
     <div data-dojo-type="dojox.mobile.SpinWheelSlot" labels="0,1,2,3,4,5,6,7,8,9" labelFrom="0" labelTo="0" value="1" style="width: 70px;"></div>
     <div data-dojo-type="dojox.mobile.SpinWheelSlot" labels="0,1,2,3,4,5,6,7,8,9" labelFrom="0" labelTo="0" value="5" style="width: 70px;"></div>
     <div data-dojo-type="dojox.mobile.SpinWheelSlot" labels="0,1,2,3,4,5,6,7,8,9" labelFrom="0" labelTo="0" value="0" style="width: 70px;"></div>
   </div>
 </div>
  <ul data-dojo-type="dojox.mobile.RoundRectList">
    <li data-dojo-type="dojox.mobile.ListItem" id="dateListItem" moveTo="detailsView_Date" rightText="2012-10-09" rightIcon="mblDomButtonArrow">
Date</li>
    <li data-dojo-type="dojox.mobile.ListItem" id="notesListItem" moveTo="detailsView_Notes" rightText="Ran 5 mi..." rightIcon="mblDomButtonArrow">
      Notes</li>
  </ul>
</div>
 <div data-dojo-type="dojox.mobile.ScrollableView" id="detailsView_Date" keepScrollPos="false" scrollBar="true">
  <h1 label="Date" data-dojo-type="dojox.mobile.Heading" back="Details" moveTo="detailsView" fixed="top"></h1>
  <div data-dojo-type="dojox.mobile.SpinWheelDatePicker" id="dateSpinWheel" style="margin-top: 10px;"></div>
</div>
 <div data-dojo-type="dojox.mobile.ScrollableView" id="detailsView_Notes" keepScrollPos="false" scrollBar="true">
  <h1 label="Notes" data-dojo-type="dojox.mobile.Heading" back="Details" moveTo="detailsView" fixed="top"></h1>
  <div data-dojo-type="dojox.mobile.RoundRect">
   <textarea data-dojo-type="dojox.mobile.TextArea" id="notesTextArea" value="Ran 5 miles and ate lots of broccoli!" style="width: 100%; height: 150px;"></textarea>
 </div>
</div>
</body>
</html>

```

The about HTML requires a json file call weights.json with the following content:

```
{
    "items": [
        {label: "149", moveTo: "detailsView", rightText: "2012-10-01"},
        {label: "150", moveTo: "detailsView", rightText: "2012-10-09"},
        {label: "151", moveTo: "detailsView", rightText: "2012-10-15"},
        {label: "148", moveTo: "detailsView", rightText: "2012-10-23"},
        {label: "146", moveTo: "detailsView", rightText: "2012-11-01"}
    ]
}
```

Once these files are created and saved, open index.html, from the states palette select `#detilsView`
Save and close the file index.html
Reopen index.html note that the middle spinWheelSlot is visually a mess
![Screen Shot 2013-01-14 at 9 02 53 PM](https://f.cloud.github.com/assets/717822/115093/d29b44aa-6bc1-11e2-88dc-6e5a35307d32.png)

**Comment by peller** [comment]
see also #3584

**Comment by aerwin** [comment]
And, note this is a somewhat big deal since it's the HTML generated by Part 1 of my Maqetta article series and a lot of people have been reading and using that tutorial.

**Comment by peller** [comment]
b84ad250bb5ac3e92de765fd7b5dbbc9c947c502 put in a log message so we can tell if the monkey patch is in place.  Perhaps there's a timing issue with dojo/ready or some condition such that the patch isn't taking hold (result: message always fires, even when widget renders incorrectly)

**Comment by peller** [comment]
experiment with a better patch in 93e94c87e085a69b4f5177529ea092978caf2136

**Comment by peller** [comment]
seems to be fixed by above patch.  Will be included in Dojo 1.9
